core/string/ustring.cpp

```diff
@@ -3942,10 +3942,12 @@ String String::format(const Variant &values, const String &placeholder) const {
 		for (int i = 0; i < values_arr.size(); i++) {
 			String i_as_str = String::num_int64(i);
 
-			if (values_arr[i].get_type() == Variant::ARRAY) { //Array in Array structure [["name","RobotGuy"],[0,"godot"],["strength",9000.91]]
+#ifndef DISABLE_DEPRECATED
+			if (values_arr[i].get_type() == Variant::ARRAY) { // Array in Array structure [["name","RobotGuy"], [0,"godot"], ["strength",9000.91]].
 				Array value_arr = values_arr[i];
 
 				if (value_arr.size() == 2) {
+					WARN_DEPRECATED_MSG("Arrays structured as key-value pairs in String.format() are deprecated. Consider using a Dictionary instead.");
```

Comment:
I think it makes sense to warn on any array size. I'd also clarify what is happening:
 
```suggestion
				if (value_arr.size() == 2) {
					WARN_DEPRECATED_MSG("When using Arrays in Arrays, they are interpreted as key-value pairs and must be of length 2. This behavior is deprecated. Consider formatting with a Dictionary instead.");
```